hy/models.py

```diff
@@ -171,18 +171,17 @@ class HyComplex(HyObject, complex):
     complex(foo) was called, given HyComplex(foo).
     """
 
-    def __new__(cls, num, *args, **kwargs):
-        value = super(HyComplex, cls).__new__(cls, strip_digit_separators(num))
-        if isinstance(num, string_types):
-            p1, _, p2 = num.lstrip("+-").replace("-", "+").partition("+")
+    def __new__(cls, real, imag=0, *args, **kwargs):
+        if isinstance(real, string_types):
+            value = super(HyComplex, cls).__new__(
+                cls, strip_digit_separators(real)
+            )
```

Comment:
You don't need a separate line for the close parenthesis. It can go at the end of the preceding line.